File: sample_codes/python/auth_api/post-v2-oauth2-refresh_token.py
import requests, json

end_point = 'https://public-api.hakaru.ai'
path = '/v2/oauth2/refresh_token'
url = end_point + path

refreshToken = 'REFRESH_TOKEN'

body = { 'refresh_token': refreshToken }
headers = { 'Content-Type': 'application/json' }

try:
    response = requests.post(url, data=json.dumps(body), headers=headers)
    response.raise_for_status()

    data = response.json()
    print('HTTP Status Code:', response.status_code)
    print('Access Token:', data['access_token'])
except requests.exceptions.RequestException as err:
    print(err)
    print('Response Body:', json.dumps(response.json(), indent=2))

File: sample_codes/php/auth_api/post-v2-oauth2-refresh_token.php
<?php
$endPoint = "https://public-api.hakaru.ai";
$path = "/v2/oauth2/refresh_token";
$url = $endPoint . $path;

$refreshToken = "REFRESH_TOKEN";
$requestBody = json_encode(array(
        "refresh_token" => $refreshToken
    ));

$ch = curl_init($url);
curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-type: application/json'));
curl_setopt($ch, CURLOPT_POST, TRUE);
curl_setopt($ch, CURLOPT_POSTFIELDS, $requestBody);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);

$strResult = curl_exec($ch);
$httpcode = curl_getinfo($ch, CURLINFO_RESPONSE_CODE);

curl_close($ch);

$result = json_decode($strResult, TRUE);

print_r("HTTP Status Code: " . $httpcode . "\n");

if ($httpcode == 200) {
    print_r("Access Token: " . $result["access_token"] . "\n");
} else {
    print_r("Response Body: " . $strResult . "\n");
}

?>